nifi-nar-bundles/nifi-standard-services/nifi-hbase_1_1_2-client-service-bundle/nifi-hbase_1_1_2-client-service/src/main/java/org/apache/nifi/hbase/HBase_1_1_2_LookupService.java

```diff
@@ -0,0 +1,168 @@
+            .name("hb-lu-return-qfs")
+            .displayName("Column Qualifiers")
+            .description("The column qualifies that will be returned.")
+            .required(false)
+            .addValidator(Validator.VALID)
+            .build();
+
+    static {
+        REQUIRED_KEYS = new HashSet<String>();
+        REQUIRED_KEYS.add("rowKey");
+    }
+
+    private String tableName;
+    private List<byte[]> families;
+    private List<byte[]> qualifiers;
+
+    private List<PropertyDescriptor> lookupProperties;
+
+    @Override
+    protected void init(ControllerServiceInitializationContext config) throws InitializationException {
+        super.init(config);
+        this.lookupProperties = new ArrayList<>();
+        this.lookupProperties.addAll(properties);
+        this.lookupProperties.add(TABLE_NAME);
+        this.lookupProperties.add(RETURN_CFS);
+        this.lookupProperties.add(RETURN_QFS);
+    }
+
+    @Override
+    public Optional lookup(Map coordinates) throws LookupFailureException {
+        byte[] rowKey = ((String)coordinates.get("rowKey")).getBytes();
+        try {
+            Map<String, Object> values = new HashMap<>();
+            try (Table table = this.connection.getTable(TableName.valueOf(tableName))) {
+                Get get = new Get(rowKey);
+                Result result = table.get(get);
+
+                for (byte[] fam : families) {
+                    NavigableMap<byte[], byte[]>  map = result.getFamilyMap(fam);
+                    for (Map.Entry<byte[], byte[]> entry : map.entrySet()) {
+                        if (qualifiers.contains(entry.getKey()) || qualifiers.size() == 0) {
+                            values.put(new String(entry.getKey()), new String(entry.getValue()));
+                        }
+                    }
+                }
+            }
+
+            if (values.size() == 1) {
+                return Optional.ofNullable(values.values().iterator().next());
+            } else if (values.size() > 1) {
+                final List<RecordField> fields = new ArrayList<>();
+                fields.add(new RecordField("key1", RecordFieldType.STRING.getDataType()));
+                fields.add(new RecordField("key2", RecordFieldType.STRING.getDataType()));
+                fields.add(new RecordField("key3", RecordFieldType.STRING.getDataType()));
+                final RecordSchema schema = new SimpleRecordSchema(fields);
+                return Optional.ofNullable(new MapRecord(schema, values));
+            } else {
+                throw new LookupFailureException(String.format("Nothing was found that matched the criteria for row key %s", coordinates.get("rowKey")));
+            }
+        } catch (IOException e) {
+            getLogger().error("Error occurred loading {}", new Object[] { coordinates.get("rowKey") }, e);
+            throw new LookupFailureException(e);
+        }
+    }
+
+    @Override
+    public Class<?> getValueType() {
+        return null;
```

Comment:
This should not be returning null. It needs to return the class for the type of object that will be returned -- namely, Record.class